woocommerce/woocommerce
# Add promoted card to in-App marketplace page

### All Submissions:






### Changes proposed in this Pull Request:

The search on `WooCommerce > Marketplace` page that shows product from the WooCommerce.com marketplace as cards. During previous re-design phase we added updated product cards in #30410.

This PR adds a new style of "promoted" product cards by:

- adding a new style card
- adding a top border to the card (the color is defined in the JSON feed)
- adding a colored button (the color is defined in the JSON feed)
- adding an automated logic that changes the button text color base on the button background-color

Closes Automattic/woocommerce.com#10885

### Screenshots

![](https://d.pr/i/LmuoA7+)

### How to test the changes in this Pull Request:

- Check out this branch.
- Edit `includes/admin/class-wc-admin-addons.php`.
- On line 66, change `$featured = get_transient( 'wc_addons_featured' );` to bypass the transient check:

```
$featured = false; // TODO - REMOVE THIS
```

- On line 83, change `https://woocommerce.com/wp-json/wccom-extensions/2.0/featured' . $parameter_string` to point to our mock WCCOM featured API data: `https://gist.githubusercontent.com/corsonr/6ac9a36583dce3a2d2d9e97f4ddf7847/raw/60a38e52de705566bdb74b2059d6acc6a086379b/featured.json`.

```php
$raw_featured = wp_safe_remote_get(
	'https://gist.githubusercontent.com/andfinally/be6f6caae34f3f3c1c5de1df0ba30681/raw/featured.json',
	array(
		'headers'    => $headers,
		'user-agent' => 'WooCommerce Addons Page',
	)
);
```

the headers could have to be removed from the `wp_safe_remote_get` call.

- On line 142, change `'https://woocommerce.com/wp-json/wccom-extensions/1.0/search' . $parameters,` to point to another mock JSON endpoint: `https://gist.githubusercontent.com/andfinally/cdea0c85da10bc037dad6aaa8bf28d0e/raw/search.json`.

```php
$raw_extensions = wp_safe_remote_get(
	'https://gist.githubusercontent.com/andfinally/cdea0c85da10bc037dad6aaa8bf28d0e/raw/search.json',
	array( 'headers' => $headers )
);
```

*  View the Featured page of WooCommerce > Marketplace on your WooCommerce site `wp-admin/admin.php?page=wc-addons&section=_featured`.
* In the `Discover our favorites` group you should see the two new styles of card shown in this screenshot:

<img width="750" alt="image" src="https://user-images.githubusercontent.com/1647564/137333981-1ee3a9e4-0069-4920-8b35-a988568382d5.png">

* View a category page like `wp-admin/admin.php?page=wc-addons&section=marketing-extensions` or search results page. You should see the two styles of card shown here at the top:

<img width="750" alt="image" src="https://user-images.githubusercontent.com/1647564/137336982-35aa77a8-ccfa-4519-84f1-87a1411a8623.png">

### Other information:




### Changelog entry

> Tweak - Add promoted cards styling to marketplace section. #30861

### FOR PR REVIEWER ONLY:
